controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTrigger.java

```diff
@@ -222,35 +222,46 @@ public boolean trigger(Job job) {
                 .map(Job::jobType).collect(toList());
     }
 
-    /**
-     * Triggers a change of this application
-     *
-     * @param applicationId the application to trigger
-     * @throws IllegalArgumentException if this application already has an ongoing change
-     */
+    /** Triggers a change of this application, unless it already has a change. */
     public void triggerChange(ApplicationId applicationId, Change change) {
         applications().lockOrThrow(applicationId, application -> {
-            if (application.get().change().isPresent() && ! application.get().deploymentJobs().hasFailures())
-                throw new IllegalArgumentException("Could not start " + change + " on " + application + ": " +
-                                                   application.get().change() + " is already in progress");
-            application = application.withChange(change);
-            if (change.application().isPresent())
-                application = application.withOutstandingChange(Change.empty());
+            if ( ! application.get().change().isPresent()) {
+                if (change.application().isPresent())
+                    application = application.withOutstandingChange(Change.empty());
 
-            applications().store(application);
+                applications().store(application.withChange(change));
+            }
+        });
+    }
+
+    /** Overrides the given application's platform and application changes with any contained in the given change. */
+    public void forceChange(ApplicationId applicationId, Change change) {
+        applications().lockOrThrow(applicationId, application -> {
+            Change current = application.get().change();
```

Comment:
"Merging" changes like this is really what you are doing in that code I commented on earlier today as well. It would be nice to move that into Change. Perhaps 3 methods, one taking a Change and two taking Optionals containing the two kinds of change. "updatedWith" is the best name suggestion I can come up with.